poppler_min: build fails on Darwin

## Issue description

Build log:
```
[11/109] Building CXX object CMakeFiles/poppler.dir/goo/grandom.cc.o
FAILED: CMakeFiles/poppler.dir/goo/grandom.cc.o 
/nix/store/vkq80zx3dmaypw02rh8kax682cr6y2jf-clang-wrapper-5.0.2/bin/clang++  -Dpoppler_EXPORTS -I../ -I../fofi -I../goo -I../poppler -I. -Ipoppler -I/nix/store/k8fv83djxw88wgbw8q3wnkh4jkqby860-fontconfig-2.12.6-dev/include -I/nix/store/j906hinjj79vjjxw66cfnalmwbfgqqw6-freetype-2.9-dev/include/freetype2 -I/nix/store/s36qv9ljd5vyq2281j02fzzwv8ga6jzq-zlib-1.2.11-dev/include -I/nix/store/pzlb556z4ch7x5ryl49gk5aicpsyrfci-libjpeg-turbo-2.0.1-dev/include -I/nix/store/kl2a5d63kjpncjllpcm3vzv15g7ykj6h-libpng-apng-1.6.35-dev/include -I/nix/store/ki349clqi68g9sy1j3jhahn14cgmrw5w-libtiff-4.0.10-dev/include -I/nix/store/dmh5qxdisbccjc02rcmhg7749g4a6dgw-openjpeg-2.3.0-dev/include/openjpeg-2.3 -I/nix/store/j4vcvphh8jcxj1hlhxjs34lrac0xdi4y-lcms2-2.9-dev/include -Wall -Wextra -Wpedantic -Wno-unused-parameter -Wcast-align -Wformat-security -Wframe-larger-than=65536 -Wmissing-format-attribute -Wnon-virtual-dtor -Woverloaded-virtual -Wmissing-declarations -Wundef -Wzero-as-null-pointer-constant -std=c++11 -O3 -DNDEBUG -arch x86_64 -fPIC -MD -MT CMakeFiles/poppler.dir/goo/grandom.cc.o -MF CMakeFiles/poppler.dir/goo/grandom.cc.o.d -o CMakeFiles/poppler.dir/goo/grandom.cc.o -c ../goo/grandom.cc
../goo/grandom.cc:19:1: error: 'auto' return without trailing return type; deduced return types are a C++14 extension
auto& grandom_engine()
^
../goo/grandom.cc:22:10: error: no viable conversion from returned value of type 'std::default_random_engine' (aka 'linear_congruential_engine<unsigned int, 48271, 0, 2147483647>') to function return type 'int'
  return engine;
         ^~~~~~
2 errors generated.
ninja: build stopped: subcommand failed.

```

#35667 might be related?

### Steps to reproduce

```
nix build -f '<nixpkgs>' poppler_min
```


## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.2.0, macOS 10.14.2`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.3`
 - channels(mnacamura): `"darwin, nixpkgs-19.03pre161900.61c3169a0e1"`
 - nixpkgs: `/Users/mnacamura/repos/nixpkgs`

--- a/pkgs/development/libraries/poppler/default.nix
+++ b/pkgs/development/libraries/poppler/default.nix
@@ -34,7 +34,7 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ cmake ninja pkgconfig ];
 
   # Not sure when and how to pass it.  It seems an upstream bug anyway.
-  CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++11";
+  CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++14";
 
   cmakeFlags = [
     (mkFlag true "XPDF_HEADERS")

